featurebyte/core/accessor/target_string.py

```diff
@@ -0,0 +1,278 @@
+"""
+Target string accessor module.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional, TypeVar
+
+from featurebyte.common.doc_util import FBAutoDoc
+from featurebyte.core.accessor.string import StringAccessor
+from featurebyte.query_graph.node.string import Side
+
+if TYPE_CHECKING:
+    from featurebyte.api.target import Target
+else:
+    Target = TypeVar("Target")
+
+
+class TargetStrAccessorMixin:
+    """
+    TargetStrAccessorMixin class
+    """
+
+    @property
+    def str(self: Target) -> StringAccessor:  # type: ignore # pylint: disable=invalid-name
+        """
+        str accessor object
+
+        Returns
+        -------
+        StringAccessor
+        """
+        return TargetStringAccessor(self)
+
+
+class TargetStringAccessor(StringAccessor):
+    """
+    TargetStringAccessor class used to manipulate string type Target objects
+    """
+
+    # documentation metadata
+    __fbautodoc__ = FBAutoDoc()
+
+    def len(self) -> Target:
+        """
+        Computes length of each string element.
+
+        Returns
+        -------
+        Target
+            A new Target object.
+
+        Examples
+        --------
+        Compute the length of each string element in the ProductGroupLookup target:
+
+        >>> target = catalog.get_target("ProductGroupLookup")  # doctest: +SKIP
```

Comment:
i've skipped these for now. will unskip when i add the lookup target functionality.